# Load Packages & Data

In [1]:
import pkg_resources
try:
    pkg_resources.get_distribution('pingouin')
except pkg_resources.DistributionNotFound:
    !pip install pingouin

In [2]:
import pandas as pd
import numpy as np
import pingouin as pg
import scipy.stats as stats
import statsmodels.stats.multicomp as multi
import statsmodels.api as sm
pd.set_option('display.max.columns', None) # display all columns
pd.set_option('display.float_format', '{:.3f}'.format)

In [3]:
df = pd.read_excel('/content/drive/MyDrive/PT Research/data/final_updated_validation_dataset_v2.1.xlsx')
df.head()

RECORD_ID  rater_1_id  rater_2_id  rater_1_R_HKAA  rater_2_R_HKAA  \
0         19           4           6         179.025         178.740   
1         30           2           6         164.255         164.602   
2         44           1           6         171.452         170.470   
3        111           4           6         179.944         179.641   
4        128           1           6         175.928         177.414   

   R_HKAA_diff_between_raters  R_HKAA_avg_of_raters  rater_1_L_HKAA  \
0                       0.285               178.882         177.659   
1                      -0.347               164.429         179.862   
2                       0.982               170.961         171.135   
3                       0.303               179.792         176.618   
4                      -1.486               176.671         170.109   

   rater_2_L_HKAA  L_HKAA_diff_between_raters  L_HKAA_avg_of_raters  \
0         177.354                       0.305               177.506   
1         179.739                       0.123               179.800   
2         170.486                       0.649               170.810   
3         177.214                      -0.596               176.916   
4         169.916                       0.193               170.012   

   R_redcap_class  R_class_w_2_degree_cutoff  R_class_w_3_degree_cutoff  \
0               1                          0                          0   
1               2                          2                          2   
2               1                          1                          1   
3               1                          1                          0   
4               1                          0                          0   

   L_redcap_class  L_class_w_2_degree_cutoff  L_class_w_3_degree_cutoff  \
0               1                          1                          0   
1               2                          0                          0   
2               1                          1                          1   
3               1                          1                          1   
4               1                          1                          1   

  DL_R_FEMORAL_HEAD_COORDS DL_R_KNEE_COORDS DL_R_ANKLE_COORDS  \
0              (793, 2463)      (717, 5009)       (692, 7202)   
1              (689, 2243)     (1016, 4879)       (678, 7206)   
2              (653, 2557)      (671, 5113)      (1023, 7367)   
3              (658, 1877)      (633, 4750)       (611, 7278)   
4              (801, 2524)      (879, 5092)       (778, 7388)   

  DL_L_FEMORAL_HEAD_COORDS DL_L_KNEE_COORDS DL_L_ANKLE_COORDS  DL_R_HKAA  \
0             (1917, 2463)     (1824, 5021)      (1820, 7210)    178.943   
1             (1706, 2163)     (1695, 4826)      (1715, 7187)    164.664   
2             (1765, 2519)     (1797, 5102)      (1448, 7356)    171.527   
3             (1778, 1879)     (1825, 4755)      (1730, 7281)    180.000   
4             (1859, 2564)     (1714, 5095)      (1951, 7357)    175.741   

   DL_L_HKAA  DL_R_CLASS  DL_L_CLASS    BMI  BMI_CLASS  LATERALITY  \
0    178.023       0.000       0.000 36.620      3.000           1   
1    179.278       2.000       0.000 27.120      1.000           0   
2    170.489       1.000       1.000 25.780      1.000           1   
3    176.910       0.000       1.000    NaN        NaN           1   
4    170.740       2.000       2.000 24.270      0.000           1   

   PROCEDURE_GROUP  PROCEDURE_LEVEL  
0                1                0  
1                1                0  
2                1                0  
3                1                0  
4                1                0

In [4]:
df[df['LATERALITY'] == 2]

RECORD_ID  rater_1_id  rater_2_id  rater_1_R_HKAA  rater_2_R_HKAA  \
26         637           4           6         176.622         176.525   
97        2302           2           6         177.227         176.794   
128        239           1           6         178.136         177.891   
134        410           1           6         175.349         174.795   

     R_HKAA_diff_between_raters  R_HKAA_avg_of_raters  rater_1_L_HKAA  \
26                        0.097               176.573         177.046   
97                        0.433               177.011         175.243   
128                       0.245               178.013         176.585   
134                       0.554               175.072         174.031   

     rater_2_L_HKAA  L_HKAA_diff_between_raters  L_HKAA_avg_of_raters  \
26          177.705                      -0.659               177.375   
97          175.867                      -0.624               175.555   
128         177.003                      -0.418               176.794   
134         175.138                      -1.107               174.584   

     R_redcap_class  R_class_w_2_degree_cutoff  R_class_w_3_degree_cutoff  \
26                1                          0                          0   
97                1                          1                          1   
128               1                          1                          1   
134               1                          1                          1   

     L_redcap_class  L_class_w_2_degree_cutoff  L_class_w_3_degree_cutoff  \
26                1                          1                          1   
97                1                          1                          1   
128               1                          1                          1   
134               1                          1                          1   

    DL_R_FEMORAL_HEAD_COORDS DL_R_KNEE_COORDS DL_R_ANKLE_COORDS  \
26               (661, 2357)      (789, 5025)       (703, 7384)   
97               (532, 2411)      (654, 5111)       (849, 7310)   
128              (649, 2160)      (795, 3641)               NaN   
134              (540, 1844)      (567, 4807)       (835, 7346)   

    DL_L_FEMORAL_HEAD_COORDS DL_L_KNEE_COORDS DL_L_ANKLE_COORDS  DL_R_HKAA  \
26              (1753, 2369)     (1711, 5030)      (1634, 7370)    175.165   
97              (1648, 2423)     (1569, 5116)      (1389, 7333)    177.520   
128                      NaN     (1650, 3546)      (1744, 5476)        NaN   
134             (1636, 1848)     (1716, 4810)      (1541, 7306)    174.497   

     DL_L_HKAA  DL_R_CLASS  DL_L_CLASS    BMI  BMI_CLASS  LATERALITY  \
26     179.020       2.000       0.000 24.550      0.000           2   
97     177.039       0.000       0.000 27.960      1.000           2   
128        NaN         NaN         NaN 18.180     -1.000           2   
134    174.442       1.000       1.000 41.300      4.000           2   

     PROCEDURE_GROUP  PROCEDURE_LEVEL  
26                 1                0  
97                 1                0  
128                0                0  
134                1                0

In [5]:
# treat rater_1 as one person, otherwise ICC can't be calculated
df['rater_1_id'] = 1
df.head()

RECORD_ID  rater_1_id  rater_2_id  rater_1_R_HKAA  rater_2_R_HKAA  \
0         19           1           6         179.025         178.740   
1         30           1           6         164.255         164.602   
2         44           1           6         171.452         170.470   
3        111           1           6         179.944         179.641   
4        128           1           6         175.928         177.414   

   R_HKAA_diff_between_raters  R_HKAA_avg_of_raters  rater_1_L_HKAA  \
0                       0.285               178.882         177.659   
1                      -0.347               164.429         179.862   
2                       0.982               170.961         171.135   
3                       0.303               179.792         176.618   
4                      -1.486               176.671         170.109   

   rater_2_L_HKAA  L_HKAA_diff_between_raters  L_HKAA_avg_of_raters  \
0         177.354                       0.305               177.506   
1         179.739                       0.123               179.800   
2         170.486                       0.649               170.810   
3         177.214                      -0.596               176.916   
4         169.916                       0.193               170.012   

   R_redcap_class  R_class_w_2_degree_cutoff  R_class_w_3_degree_cutoff  \
0               1                          0                          0   
1               2                          2                          2   
2               1                          1                          1   
3               1                          1                          0   
4               1                          0                          0   

   L_redcap_class  L_class_w_2_degree_cutoff  L_class_w_3_degree_cutoff  \
0               1                          1                          0   
1               2                          0                          0   
2               1                          1                          1   
3               1                          1                          1   
4               1                          1                          1   

  DL_R_FEMORAL_HEAD_COORDS DL_R_KNEE_COORDS DL_R_ANKLE_COORDS  \
0              (793, 2463)      (717, 5009)       (692, 7202)   
1              (689, 2243)     (1016, 4879)       (678, 7206)   
2              (653, 2557)      (671, 5113)      (1023, 7367)   
3              (658, 1877)      (633, 4750)       (611, 7278)   
4              (801, 2524)      (879, 5092)       (778, 7388)   

  DL_L_FEMORAL_HEAD_COORDS DL_L_KNEE_COORDS DL_L_ANKLE_COORDS  DL_R_HKAA  \
0             (1917, 2463)     (1824, 5021)      (1820, 7210)    178.943   
1             (1706, 2163)     (1695, 4826)      (1715, 7187)    164.664   
2             (1765, 2519)     (1797, 5102)      (1448, 7356)    171.527   
3             (1778, 1879)     (1825, 4755)      (1730, 7281)    180.000   
4             (1859, 2564)     (1714, 5095)      (1951, 7357)    175.741   

   DL_L_HKAA  DL_R_CLASS  DL_L_CLASS    BMI  BMI_CLASS  LATERALITY  \
0    178.023       0.000       0.000 36.620      3.000           1   
1    179.278       2.000       0.000 27.120      1.000           0   
2    170.489       1.000       1.000 25.780      1.000           1   
3    176.910       0.000       1.000    NaN        NaN           1   
4    170.740       2.000       2.000 24.270      0.000           1   

   PROCEDURE_GROUP  PROCEDURE_LEVEL  
0                1                0  
1                1                0  
2                1                0  
3                1                0  
4                1                0

In [6]:
# check number of patients with two surgical legs
df[df.LATERALITY == 2]

RECORD_ID  rater_1_id  rater_2_id  rater_1_R_HKAA  rater_2_R_HKAA  \
26         637           1           6         176.622         176.525   
97        2302           1           6         177.227         176.794   
128        239           1           6         178.136         177.891   
134        410           1           6         175.349         174.795   

     R_HKAA_diff_between_raters  R_HKAA_avg_of_raters  rater_1_L_HKAA  \
26                        0.097               176.573         177.046   
97                        0.433               177.011         175.243   
128                       0.245               178.013         176.585   
134                       0.554               175.072         174.031   

     rater_2_L_HKAA  L_HKAA_diff_between_raters  L_HKAA_avg_of_raters  \
26          177.705                      -0.659               177.375   
97          175.867                      -0.624               175.555   
128         177.003                      -0.418               176.794   
134         175.138                      -1.107               174.584   

     R_redcap_class  R_class_w_2_degree_cutoff  R_class_w_3_degree_cutoff  \
26                1                          0                          0   
97                1                          1                          1   
128               1                          1                          1   
134               1                          1                          1   

     L_redcap_class  L_class_w_2_degree_cutoff  L_class_w_3_degree_cutoff  \
26                1                          1                          1   
97                1                          1                          1   
128               1                          1                          1   
134               1                          1                          1   

    DL_R_FEMORAL_HEAD_COORDS DL_R_KNEE_COORDS DL_R_ANKLE_COORDS  \
26               (661, 2357)      (789, 5025)       (703, 7384)   
97               (532, 2411)      (654, 5111)       (849, 7310)   
128              (649, 2160)      (795, 3641)               NaN   
134              (540, 1844)      (567, 4807)       (835, 7346)   

    DL_L_FEMORAL_HEAD_COORDS DL_L_KNEE_COORDS DL_L_ANKLE_COORDS  DL_R_HKAA  \
26              (1753, 2369)     (1711, 5030)      (1634, 7370)    175.165   
97              (1648, 2423)     (1569, 5116)      (1389, 7333)    177.520   
128                      NaN     (1650, 3546)      (1744, 5476)        NaN   
134             (1636, 1848)     (1716, 4810)      (1541, 7306)    174.497   

     DL_L_HKAA  DL_R_CLASS  DL_L_CLASS    BMI  BMI_CLASS  LATERALITY  \
26     179.020       2.000       0.000 24.550      0.000           2   
97     177.039       0.000       0.000 27.960      1.000           2   
128        NaN         NaN         NaN 18.180     -1.000           2   
134    174.442       1.000       1.000 41.300      4.000           2   

     PROCEDURE_GROUP  PROCEDURE_LEVEL  
26                 1                0  
97                 1                0  
128                0                0  
134                1                0

There are 4 patients with two surgical legs

# Task 1: Compare reliability of annotations by right / left leg

## Calculate means

### Right leg

In [7]:
# rater 1 avg
# right leg
df['rater_1_R_HKAA'].mean().round(3)

174.17

In [8]:
# rater 2 avg
# right leg
df['rater_2_R_HKAA'].mean().round(3)

174.151

### Left leg

In [9]:
# rater 1 avg
# left leg
df['rater_1_L_HKAA'].mean().round(3)

173.63

In [10]:
# rater 2 avg
# left leg
df['rater_2_L_HKAA'].mean().round(3)

173.511

## Calculate IRRs

### Right leg

In [11]:
# right leg IRR
df_right1 = df[['RECORD_ID', 'rater_1_id', 'rater_1_R_HKAA']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_R_HKAA' : 'rater_R_HKAA' })
df_right2 = df[['RECORD_ID', 'rater_2_id', 'rater_2_R_HKAA']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_R_HKAA' : 'rater_R_HKAA' })
df_right = pd.concat([df_right1, df_right2], axis = 0)
icc_right = pg.intraclass_corr(data = df_right, targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_R_HKAA').round(3)
icc_right.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.978 91.957  189  190 0.000  [0.97, 0.98]
ICC2      Single random raters 0.978 91.510  189  189 0.000  [0.97, 0.98]
ICC3       Single fixed raters 0.978 91.510  189  189 0.000  [0.97, 0.98]
ICC1k  Average raters absolute 0.989 91.957  189  190 0.000  [0.99, 0.99]
ICC2k    Average random raters 0.989 91.510  189  189 0.000  [0.99, 0.99]
ICC3k     Average fixed raters 0.989 91.510  189  189 0.000  [0.99, 0.99]

The interrater reliability is 0.989 for the right leg.

### Left leg

In [12]:
# left leg IRR
df_left1 = df[['RECORD_ID', 'rater_1_id', 'rater_1_L_HKAA']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_L_HKAA' : 'rater_L_HKAA' })
df_left2 = df[['RECORD_ID', 'rater_2_id', 'rater_2_L_HKAA']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_L_HKAA' : 'rater_L_HKAA' })
df_left = pd.concat([df_left1, df_left2], axis = 0)
icc_left = pg.intraclass_corr(data = df_left, targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_L_HKAA').round(3)
icc_left.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.963 53.159  189  190 0.000  [0.95, 0.97]
ICC2      Single random raters 0.963 53.355  189  189 0.000  [0.95, 0.97]
ICC3       Single fixed raters 0.963 53.355  189  189 0.000  [0.95, 0.97]
ICC1k  Average raters absolute 0.981 53.159  189  190 0.000  [0.97, 0.99]
ICC2k    Average random raters 0.981 53.355  189  189 0.000  [0.97, 0.99]
ICC3k     Average fixed raters 0.981 53.355  189  189 0.000  [0.98, 0.99]

The interrater reliability is 0.981 for the left leg.

# Task 2: Compare reliability of annotations by surgical / non-surgical leg

## Calculate means

In [13]:
# LATERALITY 0 = right, 1 = left, 2 = bilateral
df_surg_right = df[(df['LATERALITY'] == 0) | (df['LATERALITY'] == 2)]
df_surg_left = df[(df['LATERALITY'] == 1) |  (df['LATERALITY'] == 2)]

### Surgical leg

In [14]:
# rater 1 avg
# surgical leg
((df_surg_left['rater_1_L_HKAA'].sum() + df_surg_right['rater_1_R_HKAA'].sum()) / (len(df_surg_left) + len(df_surg_right))).round(3)

172.457

In [15]:
# rater 2 avg
# surgical leg
((df_surg_left['rater_2_L_HKAA'].sum() + df_surg_right['rater_2_R_HKAA'].sum()) / (len(df_surg_left) + len(df_surg_right))).round(3)

172.364

### Non-surgical leg

In [16]:
# rater 1 avg
# non-surgical leg
((df_surg_left['rater_1_R_HKAA'].sum() + df_surg_right['rater_1_L_HKAA'].sum()) / (len(df_surg_left) + len(df_surg_right))).round(3)

175.441

In [17]:
# rater 2 avg
# non-surgical leg
((df_surg_left['rater_2_R_HKAA'].sum() + df_surg_right['rater_2_L_HKAA'].sum()) / (len(df_surg_left) + len(df_surg_right))).round(3)

175.407

## Calculate IRRs

### Surgical leg

In [18]:
# rater 1 surgical leg dataframe
df_surg_left1 = df_surg_left[['RECORD_ID', 'rater_1_id', 'rater_1_L_HKAA']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_L_HKAA' : 'rater_surg_HKAA' })
df_surg_right1 =  df_surg_right[['RECORD_ID', 'rater_1_id', 'rater_1_R_HKAA']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_R_HKAA' : 'rater_surg_HKAA' })
df_surg1 = pd.concat([df_surg_left1, df_surg_right1], axis = 0)

In [19]:
# rater 2 surgical leg dataframe
df_surg_left2 = df_surg_left[['RECORD_ID', 'rater_2_id', 'rater_2_L_HKAA']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_L_HKAA' : 'rater_surg_HKAA' })
df_surg_right2 =  df_surg_right[['RECORD_ID', 'rater_2_id', 'rater_2_R_HKAA']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_R_HKAA' : 'rater_surg_HKAA' })
df_surg2 = pd.concat([df_surg_left2, df_surg_right2], axis = 0)

In [20]:
# surgical leg IRR
df_surg = pd.concat([df_surg1, df_surg2], axis = 0)
icc_surg = pg.intraclass_corr(data = df_surg, targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.963 53.184  189  190 0.000  [0.95, 0.97]
ICC2      Single random raters 0.963 53.210  189  189 0.000  [0.95, 0.97]
ICC3       Single fixed raters 0.963 53.210  189  189 0.000  [0.95, 0.97]
ICC1k  Average raters absolute 0.981 53.184  189  190 0.000  [0.97, 0.99]
ICC2k    Average random raters 0.981 53.210  189  189 0.000  [0.97, 0.99]
ICC3k     Average fixed raters 0.981 53.210  189  189 0.000  [0.97, 0.99]

The interrater reliability is 0.981 for the surgical leg.

### Non-surgical leg

In [21]:
# rater 1 non-surgical leg dataframe
df_nsurg_left1 = df_surg_left[['RECORD_ID', 'rater_1_id', 'rater_1_R_HKAA']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_R_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg_right1 =  df_surg_right[['RECORD_ID', 'rater_1_id', 'rater_1_L_HKAA']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_L_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg1 = pd.concat([df_nsurg_left1, df_nsurg_right1], axis = 0)

In [22]:
# rater 2 non-surgical leg dataframe
df_nsurg_left2 = df_surg_left[['RECORD_ID', 'rater_2_id', 'rater_2_R_HKAA']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_R_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg_right2 =  df_surg_right[['RECORD_ID', 'rater_2_id', 'rater_2_L_HKAA']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_L_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg2 = pd.concat([df_nsurg_left2, df_nsurg_right2], axis = 0)

In [23]:
# non-surgical leg IRR
df_nsurg = pd.concat([df_nsurg1, df_nsurg2], axis = 0)
icc_nsurg = pg.intraclass_corr(data = df_nsurg, targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.973 74.350  189  190 0.000  [0.96, 0.98]
ICC2      Single random raters 0.973 74.112  189  189 0.000  [0.96, 0.98]
ICC3       Single fixed raters 0.973 74.112  189  189 0.000  [0.96, 0.98]
ICC1k  Average raters absolute 0.987 74.350  189  190 0.000  [0.98, 0.99]
ICC2k    Average random raters 0.987 74.112  189  189 0.000  [0.98, 0.99]
ICC3k     Average fixed raters 0.987 74.112  189  189 0.000  [0.98, 0.99]

The interrater reliability is 0.987 for the non-surgical leg.

# Task 3: Compare reliability of annotations by by surgical / non-surgical leg & weight class

In [24]:
# BMI_CLASS -1 = underweight, 0 = normal, 1 = overweight, 2 = obesity class 1, 3 = obesity class 2, 4 = obesity class 3, 5 = obesity class 4, 6 = obesity class 5
df['BMI_CLASS'].value_counts(dropna=False)

1.000     37
2.000     37
4.000     37
0.000     36
3.000     27
NaN        6
-1.000     6
5.000      4
Name: BMI_CLASS, dtype: int64

## Calculate means

### Surgical leg

#### Underweight

In [25]:
# surgical leg
# rater 1 avg
# underweight
((df_surg_left[df_surg_left['BMI_CLASS'] == -1]['rater_1_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == -1]['rater_1_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == -1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == -1]))).round(3)

174.092

In [26]:
# surgical leg
# rater 2 avg
# underweight
((df_surg_left[df_surg_left['BMI_CLASS'] == -1]['rater_2_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == -1]['rater_2_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == -1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == -1]))).round(3)

174.003

#### Normal

In [27]:
# surgical leg
# rater 1 avg
# normal
((df_surg_left[df_surg_left['BMI_CLASS'] == 0]['rater_1_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 0]['rater_1_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 0]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 0]))).round(3)

173.179

In [28]:
# surgical leg
# rater 2 avg
# normal
((df_surg_left[df_surg_left['BMI_CLASS'] == 0]['rater_2_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 0]['rater_2_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 0]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 0]))).round(3)

172.68

#### Overweight

In [29]:
# surgical leg
# rater 1 avg
# overweight
((df_surg_left[df_surg_left['BMI_CLASS'] == 1]['rater_1_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 1]['rater_1_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 1]))).round(3)

172.091

In [30]:
# surgical leg
# rater 2 avg
# overweight
((df_surg_left[df_surg_left['BMI_CLASS'] == 1]['rater_2_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 1]['rater_2_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 1]))).round(3)

172.139

#### Class 1

In [31]:
# surgical leg
# rater 1 avg
# obesity class 1
((df_surg_left[df_surg_left['BMI_CLASS'] == 2]['rater_1_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 2]['rater_1_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 2]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 2]))).round(3)

172.299

In [32]:
# surgical leg
# rater 2 avg
# obesity class 1
((df_surg_left[df_surg_left['BMI_CLASS'] == 2]['rater_2_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 2]['rater_2_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 2]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 2]))).round(3)

172.159

#### Class 2

In [33]:
# surgical leg
# rater 1 avg
# obesity class 2
((df_surg_left[df_surg_left['BMI_CLASS'] == 3]['rater_1_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 3]['rater_1_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 3]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 3]))).round(3)

172.763

In [34]:
# surgical leg
# rater 2 avg
# obesity class 2
((df_surg_left[df_surg_left['BMI_CLASS'] == 3]['rater_2_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 3]['rater_2_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 3]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 3]))).round(3)

172.645

#### Class 3

In [35]:
# surgical leg
# rater 1 avg
# obesity class 3
((df_surg_left[df_surg_left['BMI_CLASS'] == 4]['rater_1_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 4]['rater_1_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 4]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 4]))).round(3)

171.56

In [36]:
# surgical leg
# rater 2 avg
# obesity class 3
((df_surg_left[df_surg_left['BMI_CLASS'] == 4]['rater_2_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 4]['rater_2_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 4]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 4]))).round(3)

171.737

#### Class 4

In [37]:
# surgical leg
# rater 1 avg
# obesity class 4
((df_surg_left[df_surg_left['BMI_CLASS'] == 5]['rater_1_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 5]['rater_1_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 5]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 5]))).round(3)

174.272

In [38]:
# surgical leg
# rater 2 avg
# obesity class 4
((df_surg_left[df_surg_left['BMI_CLASS'] == 5]['rater_2_L_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 5]['rater_2_R_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 5]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 5]))).round(3)

174.64

### Non-surgical leg

#### Underweight

In [39]:
# non-surgical leg
# rater 1 avg
# underweight
((df_surg_left[df_surg_left['BMI_CLASS'] == -1]['rater_1_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == -1]['rater_1_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == -1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == -1]))).round(3)

177.043

In [40]:
# non-surgical leg
# rater 2 avg
# underweight
((df_surg_left[df_surg_left['BMI_CLASS'] == -1]['rater_2_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == -1]['rater_2_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == -1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == -1]))).round(3)

176.857

#### Normal

In [41]:
# non-surgical leg
# rater 1 avg
# normal
((df_surg_left[df_surg_left['BMI_CLASS'] == 0]['rater_1_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 0]['rater_1_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 0]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 0]))).round(3)

176.059

In [42]:
# non-surgical leg
# rater 2 avg
# normal
((df_surg_left[df_surg_left['BMI_CLASS'] == 0]['rater_2_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 0]['rater_2_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 0]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 0]))).round(3)

176.098

#### Overweight

In [43]:
# non-surgical leg
# rater 1 avg
# overweight
((df_surg_left[df_surg_left['BMI_CLASS'] == 1]['rater_1_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 1]['rater_1_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 1]))).round(3)

176.032

In [44]:
# non-surgical leg
# rater 2 avg
# overweight
((df_surg_left[df_surg_left['BMI_CLASS'] == 1]['rater_2_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 1]['rater_2_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 1]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 1]))).round(3)

175.888

#### Class 1

In [45]:
# non-surgical leg
# rater 1 avg
# obesity class 1
((df_surg_left[df_surg_left['BMI_CLASS'] == 2]['rater_1_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 2]['rater_1_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 2]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 2]))).round(3)

174.866

In [46]:
# non-surgical leg
# rater 2 avg
# obesity class 1
((df_surg_left[df_surg_left['BMI_CLASS'] == 2]['rater_2_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 2]['rater_2_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 2]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 2]))).round(3)

174.913

#### Class 2

In [47]:
# non-surgical leg
# rater 1 avg
# obesity class 2
((df_surg_left[df_surg_left['BMI_CLASS'] == 3]['rater_1_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 3]['rater_1_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 3]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 3]))).round(3)

172.992

In [48]:
# non-surgical leg
# rater 2 avg
# obesity class 2
((df_surg_left[df_surg_left['BMI_CLASS'] == 3]['rater_2_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 3]['rater_2_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 3]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 3]))).round(3)

172.727

#### Class 3

In [49]:
# non-surgical leg
# rater 1 avg
# obesity class 3
((df_surg_left[df_surg_left['BMI_CLASS'] == 4]['rater_1_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 4]['rater_1_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 4]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 4]))).round(3)

175.983

In [50]:
# non-surgical leg
# rater 2 avg
# obesity class 3
((df_surg_left[df_surg_left['BMI_CLASS'] == 4]['rater_2_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 4]['rater_2_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 4]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 4]))).round(3)

176.029

#### Class 4

In [51]:
# non-surgical leg
# rater 1 avg
# obesity class 4
((df_surg_left[df_surg_left['BMI_CLASS'] == 5]['rater_1_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 5]['rater_1_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 5]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 5]))).round(3)

176.764

In [52]:
# non-surgical leg
# rater 2 avg
# obesity class 4
((df_surg_left[df_surg_left['BMI_CLASS'] == 5]['rater_2_R_HKAA'].sum() +
  df_surg_right[df_surg_right['BMI_CLASS'] == 5]['rater_2_L_HKAA'].sum()) /
   (len(df_surg_left[df_surg_left['BMI_CLASS'] == 5]) +
    len(df_surg_right[df_surg_right['BMI_CLASS'] == 5]))).round(3)

176.868

## Calculate IRRs

### Surgical leg

In [53]:
# rater 1 surgical leg dataframe
df_surg_left1 = df_surg_left[['RECORD_ID', 'rater_1_id', 'rater_1_L_HKAA', 'BMI_CLASS']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_L_HKAA' : 'rater_surg_HKAA' })
df_surg_right1 =  df_surg_right[['RECORD_ID', 'rater_1_id', 'rater_1_R_HKAA', 'BMI_CLASS']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_R_HKAA' : 'rater_surg_HKAA' })
df_surg1 = pd.concat([df_surg_left1, df_surg_right1], axis = 0)

In [54]:
# rater 2 surgical leg dataframe
df_surg_left2 = df_surg_left[['RECORD_ID', 'rater_2_id', 'rater_2_L_HKAA', 'BMI_CLASS']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_L_HKAA' : 'rater_surg_HKAA' })
df_surg_right2 =  df_surg_right[['RECORD_ID', 'rater_2_id', 'rater_2_R_HKAA', 'BMI_CLASS']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_R_HKAA' : 'rater_surg_HKAA' })
df_surg2 = pd.concat([df_surg_left2, df_surg_right2], axis = 0)

In [55]:
# combined surgical leg dataframe
df_surg = pd.concat([df_surg1, df_surg2], axis = 0)

#### Underweight

In [56]:
# surgical leg
# obesity class: underweight
# IRR
icc_surg = pg.intraclass_corr(data = df_surg[df_surg['BMI_CLASS'] == -1], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC       F  df1  df2  pval        CI95%
Type                                                                     
ICC1    Single raters absolute 0.989 186.899    5    6 0.000  [0.94, 1.0]
ICC2      Single random raters 0.989 160.667    5    5 0.000  [0.93, 1.0]
ICC3       Single fixed raters 0.988 160.667    5    5 0.000  [0.91, 1.0]
ICC1k  Average raters absolute 0.995 186.899    5    6 0.000  [0.97, 1.0]
ICC2k    Average random raters 0.995 160.667    5    5 0.000  [0.96, 1.0]
ICC3k     Average fixed raters 0.994 160.667    5    5 0.000  [0.96, 1.0]

The interrater reliability is 0.995 for the surgical leg for underweight obesity class.

#### Normal

In [57]:
# surgical leg
# obesity class: normal
# IRR
icc_surg = pg.intraclass_corr(data = df_surg[df_surg['BMI_CLASS'] == 0], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.849 12.282   35   36 0.000  [0.73, 0.92]
ICC2      Single random raters 0.850 12.481   35   35 0.000  [0.73, 0.92]
ICC3       Single fixed raters 0.852 12.481   35   35 0.000  [0.73, 0.92]
ICC1k  Average raters absolute 0.919 12.282   35   36 0.000  [0.84, 0.96]
ICC2k    Average random raters 0.919 12.481   35   35 0.000  [0.84, 0.96]
ICC3k     Average fixed raters 0.920 12.481   35   35 0.000  [0.84, 0.96]

The interrater reliability is 0.919 for the surgical leg for normal obesity class.

#### Overweight

In [58]:
# surgical leg
# obesity class: overweight
# IRR
icc_surg = pg.intraclass_corr(data = df_surg[df_surg['BMI_CLASS'] == 1], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC       F  df1  df2  pval         CI95%
Type                                                                      
ICC1    Single raters absolute 0.989 173.986   36   37 0.000  [0.98, 0.99]
ICC2      Single random raters 0.989 169.966   36   36 0.000  [0.98, 0.99]
ICC3       Single fixed raters 0.988 169.966   36   36 0.000  [0.98, 0.99]
ICC1k  Average raters absolute 0.994 173.986   36   37 0.000   [0.99, 1.0]
ICC2k    Average random raters 0.994 169.966   36   36 0.000   [0.99, 1.0]
ICC3k     Average fixed raters 0.994 169.966   36   36 0.000   [0.99, 1.0]

The interrater reliability is 0.994 for the surgical leg for overweight obesity class.

#### Class 1

In [59]:
# surgical leg
# obesity class: class 1
# IRR
icc_surg = pg.intraclass_corr(data = df_surg[df_surg['BMI_CLASS'] == 2], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC       F  df1  df2  pval         CI95%
Type                                                                      
ICC1    Single raters absolute 0.989 182.181   36   37 0.000  [0.98, 0.99]
ICC2      Single random raters 0.989 183.632   36   36 0.000  [0.98, 0.99]
ICC3       Single fixed raters 0.989 183.632   36   36 0.000  [0.98, 0.99]
ICC1k  Average raters absolute 0.995 182.181   36   37 0.000   [0.99, 1.0]
ICC2k    Average random raters 0.995 183.632   36   36 0.000   [0.99, 1.0]
ICC3k     Average fixed raters 0.995 183.632   36   36 0.000   [0.99, 1.0]

The interrater reliability is 0.995 for the surgical leg for class 1 obesity class.

#### Class 2

In [60]:
# surgical leg
# obesity class: class 2
# IRR
icc_surg = pg.intraclass_corr(data = df_surg[df_surg['BMI_CLASS'] == 3], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC       F  df1  df2  pval         CI95%
Type                                                                      
ICC1    Single raters absolute 0.983 119.172   26   27 0.000  [0.96, 0.99]
ICC2      Single random raters 0.983 116.897   26   26 0.000  [0.96, 0.99]
ICC3       Single fixed raters 0.983 116.897   26   26 0.000  [0.96, 0.99]
ICC1k  Average raters absolute 0.992 119.172   26   27 0.000   [0.98, 1.0]
ICC2k    Average random raters 0.992 116.897   26   26 0.000   [0.98, 1.0]
ICC3k     Average fixed raters 0.991 116.897   26   26 0.000   [0.98, 1.0]

The interrater reliability is 0.992 for the surgical leg for class 2 obesity class.

#### Class 3

In [61]:
# surgical leg
# obesity class: class 3
# IRR
icc_surg = pg.intraclass_corr(data = df_surg[df_surg['BMI_CLASS'] == 4], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC       F  df1  df2  pval         CI95%
Type                                                                      
ICC1    Single raters absolute 0.985 128.254   36   37 0.000  [0.97, 0.99]
ICC2      Single random raters 0.985 129.899   36   36 0.000  [0.97, 0.99]
ICC3       Single fixed raters 0.985 129.899   36   36 0.000  [0.97, 0.99]
ICC1k  Average raters absolute 0.992 128.254   36   37 0.000   [0.98, 1.0]
ICC2k    Average random raters 0.992 129.899   36   36 0.000   [0.98, 1.0]
ICC3k     Average fixed raters 0.992 129.899   36   36 0.000   [0.99, 1.0]

The interrater reliability is 0.992 for the surgical leg for class 3 obesity class.

#### Class 4

In [62]:
# surgical leg
# obesity class: class 4
# IRR
icc_surg = pg.intraclass_corr(data = df_surg[df_surg['BMI_CLASS'] == 5], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_surg_HKAA').round(3)
icc_surg.set_index('Type')

Description   ICC       F  df1  df2  pval        CI95%
Type                                                                     
ICC1    Single raters absolute 0.987 152.484    3    4 0.000  [0.88, 1.0]
ICC2      Single random raters 0.987 155.368    3    3 0.001  [0.88, 1.0]
ICC3       Single fixed raters 0.987 155.368    3    3 0.001  [0.82, 1.0]
ICC1k  Average raters absolute 0.993 152.484    3    4 0.000  [0.93, 1.0]
ICC2k    Average random raters 0.993 155.368    3    3 0.001  [0.93, 1.0]
ICC3k     Average fixed raters 0.994 155.368    3    3 0.001   [0.9, 1.0]

The interrater reliability is 0.993 for the surgical leg for class 4 obesity class.

### Non-surgical leg

In [63]:
# rater 1 non-surgical leg dataframe
df_nsurg_left1 = df_surg_left[['RECORD_ID', 'rater_1_id', 'rater_1_R_HKAA', 'BMI_CLASS']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_R_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg_right1 =  df_surg_right[['RECORD_ID', 'rater_1_id', 'rater_1_L_HKAA', 'BMI_CLASS']].rename(columns = {'rater_1_id' : 'rater_id', 'rater_1_L_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg1 = pd.concat([df_nsurg_left1, df_nsurg_right1], axis = 0)

In [64]:
# rater 2 non-surgical leg dataframe
df_nsurg_left2 = df_surg_left[['RECORD_ID', 'rater_2_id', 'rater_2_R_HKAA', 'BMI_CLASS']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_R_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg_right2 =  df_surg_right[['RECORD_ID', 'rater_2_id', 'rater_2_L_HKAA', 'BMI_CLASS']].rename(columns = {'rater_2_id' : 'rater_id', 'rater_2_L_HKAA' : 'rater_nsurg_HKAA' })
df_nsurg2 = pd.concat([df_nsurg_left2, df_nsurg_right2], axis = 0)

In [65]:
# combined non-surgical leg dataframe
df_nsurg = pd.concat([df_nsurg1, df_nsurg2], axis = 0)

#### Underweight

In [66]:
# non-surgical leg
# obesity class: underweight
# IRR
icc_nsurg = pg.intraclass_corr(data = df_nsurg[df_nsurg['BMI_CLASS'] == -1], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval        CI95%
Type                                                                    
ICC1    Single raters absolute 0.968 62.331    5    6 0.000  [0.82, 1.0]
ICC2      Single random raters 0.968 60.505    5    5 0.000  [0.82, 1.0]
ICC3       Single fixed raters 0.967 60.505    5    5 0.000  [0.79, 1.0]
ICC1k  Average raters absolute 0.984 62.331    5    6 0.000   [0.9, 1.0]
ICC2k    Average random raters 0.984 60.505    5    5 0.000   [0.9, 1.0]
ICC3k     Average fixed raters 0.983 60.505    5    5 0.000  [0.88, 1.0]

The interrater reliability is 0.984 for the non-surgical leg for underweight obesity class.

#### Normal

In [67]:
# non-surgical leg
# obesity class: normal
# IRR
icc_nsurg = pg.intraclass_corr(data = df_nsurg[df_nsurg['BMI_CLASS'] == 0], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.960 49.006   35   36 0.000  [0.92, 0.98]
ICC2      Single random raters 0.960 47.707   35   35 0.000  [0.92, 0.98]
ICC3       Single fixed raters 0.959 47.707   35   35 0.000  [0.92, 0.98]
ICC1k  Average raters absolute 0.980 49.006   35   36 0.000  [0.96, 0.99]
ICC2k    Average random raters 0.980 47.707   35   35 0.000  [0.96, 0.99]
ICC3k     Average fixed raters 0.979 47.707   35   35 0.000  [0.96, 0.99]

The interrater reliability is 0.980 for the non-surgical leg for normal obesity class.

#### Overweight

In [68]:
# non-surgical leg
# obesity class: overweight
# IRR
icc_nsurg = pg.intraclass_corr(data = df_nsurg[df_nsurg['BMI_CLASS'] == 1], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.967 58.758   36   37 0.000  [0.94, 0.98]
ICC2      Single random raters 0.967 59.668   36   36 0.000  [0.94, 0.98]
ICC3       Single fixed raters 0.967 59.668   36   36 0.000  [0.94, 0.98]
ICC1k  Average raters absolute 0.983 58.758   36   37 0.000  [0.97, 0.99]
ICC2k    Average random raters 0.983 59.668   36   36 0.000  [0.97, 0.99]
ICC3k     Average fixed raters 0.983 59.668   36   36 0.000  [0.97, 0.99]

The interrater reliability is 0.983 for the non-surgical leg for overweight obesity class.

#### Class 1

In [69]:
# non-surgical leg
# obesity class: class 1
# IRR
icc_nsurg = pg.intraclass_corr(data = df_nsurg[df_nsurg['BMI_CLASS'] == 2], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.970 65.739   36   37 0.000  [0.94, 0.98]
ICC2      Single random raters 0.970 64.110   36   36 0.000  [0.94, 0.98]
ICC3       Single fixed raters 0.969 64.110   36   36 0.000  [0.94, 0.98]
ICC1k  Average raters absolute 0.985 65.739   36   37 0.000  [0.97, 0.99]
ICC2k    Average random raters 0.985 64.110   36   36 0.000  [0.97, 0.99]
ICC3k     Average fixed raters 0.984 64.110   36   36 0.000  [0.97, 0.99]

The interrater reliability is 0.985 for the non-surgical leg for class 1 obesity class.

#### Class 2

In [70]:
# non-surgical leg
# obesity class: class 2
# IRR
icc_nsurg = pg.intraclass_corr(data = df_nsurg[df_nsurg['BMI_CLASS'] == 3], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.978 88.246   26   27 0.000  [0.95, 0.99]
ICC2      Single random raters 0.978 92.421   26   26 0.000  [0.95, 0.99]
ICC3       Single fixed raters 0.979 92.421   26   26 0.000  [0.95, 0.99]
ICC1k  Average raters absolute 0.989 88.246   26   27 0.000  [0.98, 0.99]
ICC2k    Average random raters 0.989 92.421   26   26 0.000  [0.98, 0.99]
ICC3k     Average fixed raters 0.989 92.421   26   26 0.000   [0.98, 1.0]

The interrater reliability is 0.989 for the non-surgical leg for class 2 obesity class.

#### Class 3

In [71]:
# non-surgical leg
# obesity class: class 3
# IRR
icc_nsurg = pg.intraclass_corr(data = df_nsurg[df_nsurg['BMI_CLASS'] == 4], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval         CI95%
Type                                                                     
ICC1    Single raters absolute 0.976 83.828   36   37 0.000  [0.95, 0.99]
ICC2      Single random raters 0.976 81.770   36   36 0.000  [0.95, 0.99]
ICC3       Single fixed raters 0.976 81.770   36   36 0.000  [0.95, 0.99]
ICC1k  Average raters absolute 0.988 83.828   36   37 0.000  [0.98, 0.99]
ICC2k    Average random raters 0.988 81.770   36   36 0.000  [0.98, 0.99]
ICC3k     Average fixed raters 0.988 81.770   36   36 0.000  [0.98, 0.99]

The interrater reliability is 0.988 for the non-surgical leg for class 3 obesity class.

#### Class 4

In [72]:
# non-surgical leg
# obesity class: class 4
# IRR
icc_nsurg = pg.intraclass_corr(data = df_nsurg[df_nsurg['BMI_CLASS'] == 5], targets = 'RECORD_ID', raters = 'rater_id', ratings = 'rater_nsurg_HKAA').round(3)
icc_nsurg.set_index('Type')

Description   ICC      F  df1  df2  pval        CI95%
Type                                                                    
ICC1    Single raters absolute 0.957 45.611    3    4 0.001  [0.64, 1.0]
ICC2      Single random raters 0.957 35.091    3    3 0.008  [0.49, 1.0]
ICC3       Single fixed raters 0.945 35.091    3    3 0.008  [0.39, 1.0]
ICC1k  Average raters absolute 0.978 45.611    3    4 0.001  [0.78, 1.0]
ICC2k    Average random raters 0.978 35.091    3    3 0.008  [0.66, 1.0]
ICC3k     Average fixed raters 0.972 35.091    3    3 0.008  [0.56, 1.0]

The interrater reliability is 0.978 for the non-surgical leg for class 4 obesity class.

# Task 4: One-way ANOVA

## Obesity class vs average HKAA measurement

IV = obesity class

DV = HKAA measurement average between raters


### Surgical Leg

In [73]:
df_avg_surg_left = df_surg_left[['L_HKAA_avg_of_raters', 'BMI_CLASS']].rename(columns = {'L_HKAA_avg_of_raters' : 'HKAA_avg_surg'})
df_avg_surg_right = df_surg_right[['R_HKAA_avg_of_raters','BMI_CLASS']].rename(columns = {'R_HKAA_avg_of_raters' : 'HKAA_avg_surg'})
df_avg_surg = pd.concat([df_avg_surg_left, df_avg_surg_right])
df_avg_surg = df_avg_surg.dropna()
df_avg_surg

HKAA_avg_surg  BMI_CLASS
0          177.506      3.000
2          170.810      1.000
4          170.012      0.000
5          176.094      1.000
6          174.625      2.000
..             ...        ...
177        177.588      3.000
179        175.632      4.000
181        174.572      4.000
183        171.096      5.000
187        173.891      0.000

[188 rows x 2 columns]

In [74]:
# calculate average HKAA_avg_surg by BMI_CLASS
avg_by_class = df_avg_surg.groupby('BMI_CLASS')['HKAA_avg_surg'].mean().round(3)
print(avg_by_class)

BMI_CLASS
-1.000   174.047
0.000    172.930
1.000    172.115
2.000    172.229
3.000    172.704
4.000    171.649
5.000    174.456
Name: HKAA_avg_surg, dtype: float64


In [75]:
# calculate HKAA_avg_surg standard deviation by BMI_CLASS
avg_std = df_avg_surg.groupby('BMI_CLASS')['HKAA_avg_surg'].std().round(3)
print(avg_std)

BMI_CLASS
-1.000   4.489
0.000    4.375
1.000    4.848
2.000    5.069
3.000    4.753
4.000    4.920
5.000    4.423
Name: HKAA_avg_surg, dtype: float64


In [76]:
# 95% CI
lower_bound = avg_by_class - 2 * avg_std
upper_bound = avg_by_class + 2 * avg_std

# create a new dataframe to store the results
ci_df = pd.DataFrame({
    'Mean': avg_by_class,
    'Lower Bound (95% CI)': lower_bound,
    'Upper Bound (95% CI)': upper_bound,
})

print(ci_df)

             Mean  Lower Bound (95% CI)  Upper Bound (95% CI)
BMI_CLASS                                                    
-1.000    174.047               165.069               183.025
0.000     172.930               164.180               181.680
1.000     172.115               162.419               181.811
2.000     172.229               162.091               182.367
3.000     172.704               163.198               182.210
4.000     171.649               161.809               181.489
5.000     174.456               165.610               183.302


In [77]:
for index, row in ci_df.iterrows():
    print(f"{row['Mean']:.3f} [{row['Lower Bound (95% CI)']:.3f}, {row['Upper Bound (95% CI)']:.3f}]")

174.047 [165.069, 183.025]
172.930 [164.180, 181.680]
172.115 [162.419, 181.811]
172.229 [162.091, 182.367]
172.704 [163.198, 182.210]
171.649 [161.809, 181.489]
174.456 [165.610, 183.302]


In [78]:
# perform one-way ANOVA
f_value, p_value = stats.f_oneway(df_avg_surg['HKAA_avg_surg'][df_avg_surg['BMI_CLASS'] == -1],
                                  df_avg_surg['HKAA_avg_surg'][df_avg_surg['BMI_CLASS'] == 0],
                                  df_avg_surg['HKAA_avg_surg'][df_avg_surg['BMI_CLASS'] == 1],
                                  df_avg_surg['HKAA_avg_surg'][df_avg_surg['BMI_CLASS'] == 2],
                                  df_avg_surg['HKAA_avg_surg'][df_avg_surg['BMI_CLASS'] == 3],
                                  df_avg_surg['HKAA_avg_surg'][df_avg_surg['BMI_CLASS'] == 4],)
print('F value:', f_value.round(3))
print('P value:', p_value.round(3))

F value: 0.499
P value: 0.777


There's no statistically significant difference (p > 0.05) in the mean HKAA_avg_surg between the different BMI_CLASS groups.

In [79]:
# post hoc test (Tukey HSD)
posthoc = multi.MultiComparison(df_avg_surg['HKAA_avg_surg'], df_avg_surg['BMI_CLASS'])
result = posthoc.tukeyhsd()
tukey_result_df = pd.DataFrame(data=result._results_table.data[1:], columns=result._results_table.data[0])
tukey_result_df

group1  group2  meandiff  p-adj  lower  upper  reject
0   -1.000   0.000    -1.118  0.998 -7.000  4.764   False
1   -1.000   1.000    -1.933  0.957 -7.802  3.937   False
2   -1.000   2.000    -1.818  0.969 -7.700  4.064   False
3   -1.000   3.000    -1.344  0.994 -7.397  4.709   False
4   -1.000   4.000    -2.399  0.886 -8.269  3.471   False
5   -1.000   5.000     0.408  1.000 -8.537  9.353   False
6    0.000   1.000    -0.815  0.990 -4.111  2.481   False
7    0.000   2.000    -0.700  0.996 -4.018  2.618   False
8    0.000   3.000    -0.226  1.000 -3.838  3.386   False
9    0.000   4.000    -1.281  0.908 -4.577  2.015   False
10   0.000   5.000     1.526  0.997 -5.985  9.037   False
11   1.000   2.000     0.115  1.000 -3.182  3.411   False
12   1.000   3.000     0.589  0.999 -3.003  4.181   False
13   1.000   4.000    -0.466  1.000 -3.740  2.808   False
14   1.000   5.000     2.341  0.967 -5.161  9.842   False
15   2.000   3.000     0.474  1.000 -3.138  4.087   False
16   2.000   4.000    -0.581  0.999 -3.877  2.715   False
17   2.000   5.000     2.226  0.975 -5.285  9.738   False
18   3.000   4.000    -1.055  0.976 -4.647  2.537   False
19   3.000   5.000     1.752  0.993 -5.894  9.398   False
20   4.000   5.000     2.807  0.923 -4.695 10.309   False

### Non-surgical Leg

In [80]:
df_avg_nsurg_left = df_surg_left[['R_HKAA_avg_of_raters', 'BMI_CLASS']].rename(columns = {'R_HKAA_avg_of_raters' : 'HKAA_avg_nsurg'})
df_avg_nsurg_right = df_surg_right[['L_HKAA_avg_of_raters','BMI_CLASS']].rename(columns = {'L_HKAA_avg_of_raters' : 'HKAA_avg_nsurg'})
df_avg_nsurg = pd.concat([df_avg_nsurg_left, df_avg_nsurg_right])
df_avg_nsurg = df_avg_nsurg.dropna()
df_avg_nsurg

HKAA_avg_nsurg  BMI_CLASS
0           178.882      3.000
2           170.961      1.000
4           176.671      0.000
5           174.843      1.000
6           179.252      2.000
..              ...        ...
177         167.871      3.000
179         177.773      4.000
181         177.636      4.000
183         177.161      5.000
187         179.369      0.000

[188 rows x 2 columns]

In [81]:
# calculate average HKAA_avg_nsurg by BMI_CLASS
avg_by_class = df_avg_nsurg.groupby('BMI_CLASS')['HKAA_avg_nsurg'].mean().round(3)
print(avg_by_class)

BMI_CLASS
-1.000   176.950
0.000    176.079
1.000    175.960
2.000    174.889
3.000    172.860
4.000    176.006
5.000    176.816
Name: HKAA_avg_nsurg, dtype: float64


In [82]:
# calculate HKAA_avg_nsurg standard deviation by BMI_CLASS
avg_std = df_avg_nsurg.groupby('BMI_CLASS')['HKAA_avg_nsurg'].std().round(3)
print(avg_std)

BMI_CLASS
-1.000   2.249
0.000    3.086
1.000    2.774
2.000    3.975
3.000    4.375
4.000    3.533
5.000    2.230
Name: HKAA_avg_nsurg, dtype: float64


In [83]:
# 95% CI
lower_bound = avg_by_class - 2 * avg_std
upper_bound = avg_by_class + 2 * avg_std

# create a new dataframe to store the results
ci_df = pd.DataFrame({
    'Mean': avg_by_class,
    'Lower Bound (95% CI)': lower_bound,
    'Upper Bound (95% CI)': upper_bound,
})

print(ci_df)

             Mean  Lower Bound (95% CI)  Upper Bound (95% CI)
BMI_CLASS                                                    
-1.000    176.950               172.452               181.448
0.000     176.079               169.907               182.251
1.000     175.960               170.412               181.508
2.000     174.889               166.939               182.839
3.000     172.860               164.110               181.610
4.000     176.006               168.940               183.072
5.000     176.816               172.356               181.276


In [84]:
for index, row in ci_df.iterrows():
    print(f"{row['Mean']:.3f} [{row['Lower Bound (95% CI)']:.3f}, {row['Upper Bound (95% CI)']:.3f}]")

176.950 [172.452, 181.448]
176.079 [169.907, 182.251]
175.960 [170.412, 181.508]
174.889 [166.939, 182.839]
172.860 [164.110, 181.610]
176.006 [168.940, 183.072]
176.816 [172.356, 181.276]


In [85]:
# perform one-way ANOVA
f_value, p_value = stats.f_oneway(df_avg_nsurg['HKAA_avg_nsurg'][df_avg_nsurg['BMI_CLASS'] == -1],
                                  df_avg_nsurg['HKAA_avg_nsurg'][df_avg_nsurg['BMI_CLASS'] == 0],
                                  df_avg_nsurg['HKAA_avg_nsurg'][df_avg_nsurg['BMI_CLASS'] == 1],
                                  df_avg_nsurg['HKAA_avg_nsurg'][df_avg_nsurg['BMI_CLASS'] == 2],
                                  df_avg_nsurg['HKAA_avg_nsurg'][df_avg_nsurg['BMI_CLASS'] == 3],
                                  df_avg_nsurg['HKAA_avg_nsurg'][df_avg_nsurg['BMI_CLASS'] == 4])
print('F value:', f_value.round(3))
print('P value:', p_value.round(3))

F value: 3.963
P value: 0.002


There's statistically significant difference (p < 0.05) in the mean HKAA_avg_nsurg between the different BMI_CLASS groups.

In [86]:
# post hoc test (Tukey Hstd)
posthoc = multi.MultiComparison(df_avg_nsurg['HKAA_avg_nsurg'], df_avg_nsurg['BMI_CLASS'])
result = posthoc.tukeyhsd()
tukey_result_df = pd.DataFrame(data=result._results_table.data[1:], columns=result._results_table.data[0])
tukey_result_df

group1  group2  meandiff  p-adj  lower  upper  reject
0   -1.000   0.000    -0.871  0.997 -5.157  3.415   False
1   -1.000   1.000    -0.990  0.993 -5.267  3.287   False
2   -1.000   2.000    -2.061  0.783 -6.347  2.225   False
3   -1.000   3.000    -4.090  0.089 -8.501  0.320   False
4   -1.000   4.000    -0.944  0.995 -5.221  3.333   False
5   -1.000   5.000    -0.134  1.000 -6.652  6.383   False
6    0.000   1.000    -0.119  1.000 -2.521  2.283   False
7    0.000   2.000    -1.190  0.764 -3.607  1.228   False
8    0.000   3.000    -3.219  0.006 -5.851 -0.587    True
9    0.000   4.000    -0.073  1.000 -2.474  2.329   False
10   0.000   5.000     0.737  1.000 -4.736  6.210   False
11   1.000   2.000    -1.071  0.837 -3.472  1.331   False
12   1.000   3.000    -3.100  0.009 -5.718 -0.483    True
13   1.000   4.000     0.046  1.000 -2.339  2.432   False
14   1.000   5.000     0.856  0.999 -4.610  6.322   False
15   2.000   3.000    -2.030  0.250 -4.662  0.602   False
16   2.000   4.000     1.117  0.808 -1.285  3.519   False
17   2.000   5.000     1.927  0.942 -3.546  7.400   False
18   3.000   4.000     3.147  0.008  0.529  5.764    True
19   3.000   5.000     3.956  0.347 -1.615  9.527   False
20   4.000   5.000     0.810  0.999 -4.656  6.276   False

There's a statistically significant difference (p < 0.05) in the mean 'HKAA_avg_nsurg' between 'BMI_CLASS' normal & class 2, overweight & class 2, and class 2 & class 3.

## Obesity class vs difference of HKAA measurement

IV = obesity class

DV = HKAA measurement difference between raters


### Surgical Leg

In [87]:
df_diff_surg_left = df_surg_left[['L_HKAA_diff_between_raters', 'BMI_CLASS']].rename(columns = {'L_HKAA_diff_between_raters' : 'HKAA_diff_surg'})
df_diff_surg_right = df_surg_right[['R_HKAA_diff_between_raters','BMI_CLASS']].rename(columns = {'R_HKAA_diff_between_raters' : 'HKAA_diff_surg'})
df_diff_surg = pd.concat([df_diff_surg_left, df_diff_surg_right])
df_diff_surg = df_diff_surg.dropna()
df_diff_surg

HKAA_diff_surg  BMI_CLASS
0             0.305      3.000
2             0.649      1.000
4             0.193      0.000
5             0.086      1.000
6            -0.270      2.000
..              ...        ...
177           1.184      3.000
179          -0.700      4.000
181          -0.877      4.000
183           0.003      5.000
187          -0.145      0.000

[188 rows x 2 columns]

In [88]:
# calculate average HKAA_diff_surg by BMI_CLASS
avg_by_class = df_diff_surg.groupby('BMI_CLASS')['HKAA_diff_surg'].mean().round(3)
print(avg_by_class)

BMI_CLASS
-1.000    0.089
0.000     0.499
1.000    -0.048
2.000     0.140
3.000     0.118
4.000    -0.176
5.000    -0.368
Name: HKAA_diff_surg, dtype: float64


In [89]:
# calculate HKAA_diff_surg standard deviation by BMI_CLASS
avg_std = df_diff_surg.groupby('BMI_CLASS')['HKAA_diff_surg'].std().round(3)
print(avg_std)

BMI_CLASS
-1.000   0.698
0.000    2.451
1.000    0.746
2.000    0.748
3.000    0.879
4.000    0.880
5.000    0.710
Name: HKAA_diff_surg, dtype: float64


In [90]:
# 95% CI
lower_bound = avg_by_class - 2 * avg_std
upper_bound = avg_by_class + 2 * avg_std

# create a new dataframe to store the results
ci_df = pd.DataFrame({
    'Mean': avg_by_class,
    'Lower Bound (95% CI)': lower_bound,
    'Upper Bound (95% CI)': upper_bound,
})

print(ci_df)

            Mean  Lower Bound (95% CI)  Upper Bound (95% CI)
BMI_CLASS                                                   
-1.000     0.089                -1.307                 1.485
0.000      0.499                -4.403                 5.401
1.000     -0.048                -1.540                 1.444
2.000      0.140                -1.356                 1.636
3.000      0.118                -1.640                 1.876
4.000     -0.176                -1.936                 1.584
5.000     -0.368                -1.788                 1.052


In [91]:
for index, row in ci_df.iterrows():
    print(f"{row['Mean']:.3f} [{row['Lower Bound (95% CI)']:.3f}, {row['Upper Bound (95% CI)']:.3f}]")

0.089 [-1.307, 1.485]
0.499 [-4.403, 5.401]
-0.048 [-1.540, 1.444]
0.140 [-1.356, 1.636]
0.118 [-1.640, 1.876]
-0.176 [-1.936, 1.584]
-0.368 [-1.788, 1.052]


In [92]:
# perform one-way ANOVA
f_value, p_value = stats.f_oneway(df_diff_surg['HKAA_diff_surg'][df_diff_surg['BMI_CLASS'] == -1],
                                  df_diff_surg['HKAA_diff_surg'][df_diff_surg['BMI_CLASS'] == 0],
                                  df_diff_surg['HKAA_diff_surg'][df_diff_surg['BMI_CLASS'] == 1],
                                  df_diff_surg['HKAA_diff_surg'][df_diff_surg['BMI_CLASS'] == 2],
                                  df_diff_surg['HKAA_diff_surg'][df_diff_surg['BMI_CLASS'] == 3],
                                  df_diff_surg['HKAA_diff_surg'][df_diff_surg['BMI_CLASS'] == 4],)
print('F value:', f_value.round(3))
print('P value:', p_value.round(3))

F value: 1.117
P value: 0.353


There's no statistically significant difference (p > 0.05) in the mean HKAA_diff_surg between the different BMI_CLASS groups.

In [93]:
# post hoc test (Tukey HSD)
posthoc = multi.MultiComparison(df_diff_surg['HKAA_diff_surg'], df_diff_surg['BMI_CLASS'])
result = posthoc.tukeyhsd()
tukey_result_df = pd.DataFrame(data=result._results_table.data[1:], columns=result._results_table.data[0])
tukey_result_df

group1  group2  meandiff  p-adj  lower  upper  reject
0   -1.000   0.000     0.409  0.988 -1.199  2.018   False
1   -1.000   1.000    -0.137  1.000 -1.742  1.468   False
2   -1.000   2.000     0.051  1.000 -1.558  1.659   False
3   -1.000   3.000     0.029  1.000 -1.627  1.684   False
4   -1.000   4.000    -0.266  0.999 -1.871  1.340   False
5   -1.000   5.000    -0.457  0.998 -2.904  1.989   False
6    0.000   1.000    -0.546  0.545 -1.448  0.355   False
7    0.000   2.000    -0.359  0.901 -1.266  0.549   False
8    0.000   3.000    -0.381  0.912 -1.369  0.607   False
9    0.000   4.000    -0.675  0.283 -1.576  0.227   False
10   0.000   5.000    -0.867  0.870 -2.921  1.188   False
11   1.000   2.000     0.188  0.996 -0.714  1.089   False
12   1.000   3.000     0.166  0.999 -0.817  1.148   False
13   1.000   4.000    -0.128  1.000 -1.024  0.767   False
14   1.000   5.000    -0.320  0.999 -2.372  1.732   False
15   2.000   3.000    -0.022  1.000 -1.010  0.966   False
16   2.000   4.000    -0.316  0.942 -1.218  0.585   False
17   2.000   5.000    -0.508  0.990 -2.562  1.546   False
18   3.000   4.000    -0.294  0.973 -1.277  0.688   False
19   3.000   5.000    -0.486  0.993 -2.577  1.605   False
20   4.000   5.000    -0.192  1.000 -2.243  1.860   False

### Non-surgical

In [94]:
df_diff_nsurg_left = df_surg_left[['R_HKAA_diff_between_raters', 'BMI_CLASS']].rename(columns = {'R_HKAA_diff_between_raters' : 'HKAA_diff_nsurg'})
df_diff_nsurg_right = df_surg_right[['L_HKAA_diff_between_raters','BMI_CLASS']].rename(columns = {'L_HKAA_diff_between_raters' : 'HKAA_diff_nsurg'})
df_diff_nsurg = pd.concat([df_diff_nsurg_left, df_diff_nsurg_right])
df_diff_nsurg = df_diff_nsurg.dropna()
df_diff_nsurg

HKAA_diff_nsurg  BMI_CLASS
0              0.285      3.000
2              0.982      1.000
4             -1.486      0.000
5              0.850      1.000
6              0.253      2.000
..               ...        ...
177            0.658      3.000
179            0.777      4.000
181            0.319      4.000
183           -0.434      5.000
187           -0.569      0.000

[188 rows x 2 columns]

In [95]:
# calculate average HKAA_diff_nsurg by BMI_CLASS
avg_by_class = df_diff_nsurg.groupby('BMI_CLASS')['HKAA_diff_nsurg'].mean().round(3)
print(avg_by_class)

BMI_CLASS
-1.000    0.186
0.000    -0.039
1.000     0.143
2.000    -0.047
3.000     0.264
4.000    -0.046
5.000    -0.105
Name: HKAA_diff_nsurg, dtype: float64


In [96]:
# calculate HKAA_diff_nsurg standard deviation by BMI_CLASS
avg_std = df_diff_nsurg.groupby('BMI_CLASS')['HKAA_diff_nsurg'].std().round(3)
print(avg_std)

BMI_CLASS
-1.000   0.612
0.000    0.898
1.000    0.728
2.000    0.993
3.000    0.910
4.000    0.805
5.000    0.753
Name: HKAA_diff_nsurg, dtype: float64


In [97]:
# 95% CI
lower_bound = avg_by_class - 2 * avg_std
upper_bound = avg_by_class + 2 * avg_std

# create a new dataframe to store the results
ci_df = pd.DataFrame({
    'Mean': avg_by_class,
    'Lower Bound (95% CI)': lower_bound,
    'Upper Bound (95% CI)': upper_bound,
})

print(ci_df)

            Mean  Lower Bound (95% CI)  Upper Bound (95% CI)
BMI_CLASS                                                   
-1.000     0.186                -1.038                 1.410
0.000     -0.039                -1.835                 1.757
1.000      0.143                -1.313                 1.599
2.000     -0.047                -2.033                 1.939
3.000      0.264                -1.556                 2.084
4.000     -0.046                -1.656                 1.564
5.000     -0.105                -1.611                 1.401


In [98]:
for index, row in ci_df.iterrows():
    print(f"{row['Mean']:.3f} [{row['Lower Bound (95% CI)']:.3f}, {row['Upper Bound (95% CI)']:.3f}]")

0.186 [-1.038, 1.410]
-0.039 [-1.835, 1.757]
0.143 [-1.313, 1.599]
-0.047 [-2.033, 1.939]
0.264 [-1.556, 2.084]
-0.046 [-1.656, 1.564]
-0.105 [-1.611, 1.401]


In [99]:
# perform one-way ANOVA
f_value, p_value = stats.f_oneway(df_diff_nsurg['HKAA_diff_nsurg'][df_diff_nsurg['BMI_CLASS'] == -1],
                                  df_diff_nsurg['HKAA_diff_nsurg'][df_diff_nsurg['BMI_CLASS'] == 0],
                                  df_diff_nsurg['HKAA_diff_nsurg'][df_diff_nsurg['BMI_CLASS'] == 1],
                                  df_diff_nsurg['HKAA_diff_nsurg'][df_diff_nsurg['BMI_CLASS'] == 2],
                                  df_diff_nsurg['HKAA_diff_nsurg'][df_diff_nsurg['BMI_CLASS'] == 3],
                                  df_diff_nsurg['HKAA_diff_nsurg'][df_diff_nsurg['BMI_CLASS'] == 4])
print('F value:', f_value.round(3))
print('P value:', p_value.round(3))

F value: 0.726
P value: 0.605


There's statistically significant difference (p < 0.05) in the mean HKAA_diff_nsurg between the different BMI_CLASS groups.

In [100]:
# post hoc test (Tukey HSD)
posthoc = multi.MultiComparison(df_diff_nsurg['HKAA_diff_nsurg'], df_diff_nsurg['BMI_CLASS'])
result = posthoc.tukeyhsd()
tukey_result_df = pd.DataFrame(data=result._results_table.data[1:], columns=result._results_table.data[0])
tukey_result_df

group1  group2  meandiff  p-adj  lower  upper  reject
0   -1.000   0.000    -0.225  0.996 -1.280  0.831   False
1   -1.000   1.000    -0.042  1.000 -1.096  1.011   False
2   -1.000   2.000    -0.233  0.995 -1.288  0.823   False
3   -1.000   3.000     0.079  1.000 -1.008  1.165   False
4   -1.000   4.000    -0.231  0.995 -1.285  0.822   False
5   -1.000   5.000    -0.290  0.998 -1.896  1.315   False
6    0.000   1.000     0.182  0.969 -0.409  0.774   False
7    0.000   2.000    -0.008  1.000 -0.604  0.588   False
8    0.000   3.000     0.303  0.804 -0.345  0.952   False
9    0.000   4.000    -0.006  1.000 -0.598  0.585   False
10   0.000   5.000    -0.066  1.000 -1.414  1.282   False
11   1.000   2.000    -0.191  0.962 -0.782  0.401   False
12   1.000   3.000     0.121  0.998 -0.524  0.766   False
13   1.000   4.000    -0.189  0.962 -0.777  0.399   False
14   1.000   5.000    -0.248  0.998 -1.595  1.098   False
15   2.000   3.000     0.311  0.784 -0.337  0.960   False
16   2.000   4.000     0.002  1.000 -0.590  0.593   False
17   2.000   5.000    -0.058  1.000 -1.406  1.290   False
18   3.000   4.000    -0.310  0.783 -0.955  0.335   False
19   3.000   5.000    -0.369  0.985 -1.741  1.003   False
20   4.000   5.000    -0.059  1.000 -1.406  1.287   False

# Task 5: Pearson chi-square test

## Surgical leg varus / valgus / neutral rates vs obesity classes

### 3 degree cutoff

In [101]:
# deformity class dataframe
df_dclass_surg_left = df_surg_left[['L_class_w_3_degree_cutoff', 'BMI_CLASS']].rename(columns = {'L_class_w_3_degree_cutoff' : 'class_w_3_degree_cutoff'})
df_dclass_surg_right = df_surg_right[['R_class_w_3_degree_cutoff','BMI_CLASS']].rename(columns = {'R_class_w_3_degree_cutoff' : 'class_w_3_degree_cutoff'})
df_dclass_surg = pd.concat([df_dclass_surg_left, df_dclass_surg_right])
df_dclass_surg = df_dclass_surg.dropna()
df_dclass_surg

class_w_3_degree_cutoff  BMI_CLASS
0                          0      3.000
2                          1      1.000
4                          1      0.000
5                          2      1.000
6                          2      2.000
..                       ...        ...
177                        2      3.000
179                        0      4.000
181                        1      4.000
183                        1      5.000
187                        1      0.000

[188 rows x 2 columns]

In [102]:
# create a contingency table
contingency_table = pd.crosstab(df_dclass_surg['BMI_CLASS'], df_dclass_surg['class_w_3_degree_cutoff'])

print(contingency_table)

# perform chi-square test
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)

print(f"Chi-square: {chi2.round(3)}")
print(f"P value: {p.round(3)}")

class_w_3_degree_cutoff   0   1  2
BMI_CLASS                         
-1.000                    1   5  1
0.000                    13  21  3
1.000                     8  23  7
2.000                     8  23  6
3.000                    15   8  4
4.000                    14  20  4
5.000                     2   2  0
Chi-square: 15.415
P value: 0.22


There is no statistically significant (p > 0.05) association between the BMI_CLASS and class_w_3_degree_cutoff.

### 2 degree cutoff

In [103]:
# deformity class datafram
df_dclass_surg_left = df_surg_left[['L_class_w_2_degree_cutoff', 'BMI_CLASS']].rename(columns = {'L_class_w_2_degree_cutoff' : 'class_w_2_degree_cutoff'})
df_dclass_surg_right = df_surg_right[['R_class_w_2_degree_cutoff','BMI_CLASS']].rename(columns = {'R_class_w_2_degree_cutoff' : 'class_w_2_degree_cutoff'})
df_dclass_surg = pd.concat([df_dclass_surg_left, df_dclass_surg_right])
df_dclass_surg = df_dclass_surg.dropna()
df_dclass_surg

class_w_2_degree_cutoff  BMI_CLASS
0                          1      3.000
2                          1      1.000
4                          1      0.000
5                          2      1.000
6                          2      2.000
..                       ...        ...
177                        2      3.000
179                        0      4.000
181                        1      4.000
183                        1      5.000
187                        1      0.000

[188 rows x 2 columns]

In [104]:
# create a contingency table
contingency_table = pd.crosstab(df_dclass_surg['BMI_CLASS'], df_dclass_surg['class_w_2_degree_cutoff'])

print(contingency_table)

# perform chi-square test
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)

print(f"Chi-square statistic: {chi2.round(3)}")
print(f"P value: {p.round(3)}")

class_w_2_degree_cutoff  0   1  2
BMI_CLASS                        
-1.000                   0   5  2
0.000                    5  26  6
1.000                    4  27  7
2.000                    6  25  6
3.000                    9  14  4
4.000                    5  26  7
5.000                    1   3  0
Chi-square statistic: 10.03
P value: 0.613


There is no statistically significant (p > 0.05) association between the BMI_CLASS and class_w_3_degree_cutoff.